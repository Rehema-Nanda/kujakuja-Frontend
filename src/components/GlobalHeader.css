.global-header {
    background-color: #ffc300;
    z-index: 1049;
}

.navbar {
    min-height: 104px;
    border-top-width: 6px;
    border-top-style: solid;
}

.logo-parent-div{
    float: left;
}

.customer-logo-parent-div{
    float: left;
    height: 68px;
    width: 80px;
    margin-left: -16px;
    padding: 12px 12px 11px 12px;
    border-radius: 0px 8px 8px 0px;
    text-align: center;
}

.customer-logo {
    max-width: 100px;
    max-height: 100%;
    margin: auto;
}

.custom-logo{
    position: relative;
    top: 50%;
    transform: translateY(-50%);
}

.customer-logo-nav-link {
    padding: 0;
    display: inline-block;
    height: 100%;
}

.kuja-kuja-logo-nav-link {
    padding: 12px 0 0 12px;
    float: left;
}

.customer-logo-nav-link span.title-text {
    font-family: 'Nunito Sans', sans-serif !important;
    padding-left: 10px;
    color: #000000;
    font-size: 16px;
}

.navbar-toggler {
    padding: 0;
    margin-right: 16px;
    border: none;
    outline: none;
}

.global-header .navbar-toggler-icon {
    width: 32px;
    margin-right: 0;
    background-image: url('../img/hamburger.svg');
    background-repeat: no-repeat;
    background-size: 100% auto;
}

.powered-by {
    font-family: 'Nunito Sans', sans-serif !important;
    background-color: rgb(255, 195, 0);
    height: 24px;
    font-size: 11px;
    padding: 5px 5px 5px 10px;
    font-weight: 900;
    border-bottom: 1px solid #e5b200;
    color: #000000;
}

nav .navbar-nav li a {
    font-family: 'Nunito Sans', sans-serif;
    color: #171D21 !important;
    font-size: 16px;
    margin-right: 16px;
}

nav .navbar-nav li a.navbar_active {
    border-bottom: #171D21 1px solid;
}

nav .collapse.show .navbar-nav {
    margin-top: 32px;
    margin-bottom: 32px;
}

nav .collapse.show .navbar-nav li {
    padding-left: 24px;
}

nav .collapse.show .navbar-nav li a {
    margin: 16px;
    padding: 0;
}

.dropdown {
    padding-top: 8px;
    cursor: pointer;
    font-family: 'Nunito Sans', sans-serif;
    font-size: 16px;
    margin-right: 16px;
}

.dropdown-menu {
    left: -80px;
    padding: 20px;
    min-width: 200px;
    border-color: #FFFFFF;
    box-shadow: 0 10px 12px -6px rgba(12, 16, 19, 0.24);
    border-radius: 8px;
}

.dropdown-item.active, .dropdown-item:active {
    background-color: #000000;
}

.arrow-up {
    width: 0;
    height: 0;
    position: absolute;
    top: -7px;
    left: 117px;
    border-style: solid;
    border-width: 0 20px 20px 20px;
    border-color: transparent transparent #FFFFFF transparent;
    padding: 0 !important;
}

@media screen and (min-width: 992px) {
    .dropdown-menu {
        left: -150px;
    }
}

@media screen and (max-width: 992px) {
    .dropdown {
        padding-left: 16px;
    }

    .arrow-up {
        left: 12px;
    }
}
