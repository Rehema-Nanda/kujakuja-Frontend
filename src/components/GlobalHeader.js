import React from "react";
import { NavLink as Link } from "react-router-dom";
import {
    Collapse, Nav, Navbar, NavbarToggler, NavItem, NavLink, Dropdown,
    DropdownMenu,
    DropdownToggle,
    DropdownItem,
} from "reactstrap";

import "./GlobalHeader.css";
import reactCSS from "reactcss";
import i18n from "i18next";
import AppConfig from "../AppConfig";
import { AuthContext } from "../AuthContext";
import lngOptions from "../i18n/lngOptions";

export default class GlobalHeader extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isNavOpen: false,
            siteHeader: {
                highlight_colour: AppConfig.SITE_HEADER_DEFAULT_HIGHLIGHT_COLOUR,
                title_text: AppConfig.SITE_HEADER_DEFAULT_TITLE_TEXT,
                logo_url: null,
                kujakuja_logo_url: require("../img/global-logo-dark.svg"),
            },
            isLngSelectDropdownOpen: false,
        };
        this.toggleNav = this.toggleNav.bind(this);
        this.toggleLngSelectDropdown = this.toggleLngSelectDropdown.bind(this);
    }

    componentDidMount = async () => {
        await this.getSiteHeader();
    };

    generateAdminUrl = () => {
        const { location: { protocol, host } } = window;
        return `${protocol}//admin.${host}/`;
    }

    getSiteHeader = async () => {
        // retrieve site config
        const config = await this.context.makeHttpCall("config", undefined, undefined, true);

        if (config && config.data) {
            const siteHeader = { ...config.data.site_header };

            // set site favicon if there is one in the config
            if (siteHeader.favicon_url) {
                const link = document.querySelector("link[rel*='icon']") || document.createElement("link");
                link.type = "image/x-icon";
                link.rel = "shortcut icon";
                link.href = siteHeader.favicon_url;
                document.getElementsByTagName("head")[0].appendChild(link);
            }

            this.setState({
                siteHeader: {
                    ...this.state.siteHeader,
                    highlight_colour: siteHeader.highlight_colour || AppConfig.SITE_HEADER_DEFAULT_HIGHLIGHT_COLOUR,
                    title_text: siteHeader.title_text || AppConfig.SITE_HEADER_DEFAULT_TITLE_TEXT,
                    logo_url: siteHeader.logo_url,
                },
            });
        }
    };

    toggleNav() {
        const { isNavOpen } = this.state;

        // close menu if mobile
        if (window.innerWidth <= 992) {
            this.setState({
                isNavOpen: !isNavOpen,
            });
        }
    }

    toggleLngSelectDropdown() {
        const { isLngSelectDropdownOpen } = this.state;

        this.setState({
            isLngSelectDropdownOpen: !isLngSelectDropdownOpen,
        });
    }

    render() {
        const { changeLanguage, selectedLanguage } = this.props;
        const { siteHeader, isLngSelectDropdownOpen, isNavOpen } = this.state;
        const { isAuthenticated, logout } = this.context;
        const selectedLanguageOption = lngOptions.find((lng) => {
            return lng.value === selectedLanguage;
        });
        let selectedLanguageName;
        if (selectedLanguageOption) {
            selectedLanguageName = selectedLanguageOption.name;
        }
        const styles = reactCSS({
            default: {
                customHeaderBg: {
                    borderTopColor: siteHeader.highlight_colour || AppConfig.SITE_HEADER_DEFAULT_HIGHLIGHT_COLOUR,
                },
                customLogoParentBg: {
                    backgroundColor: siteHeader.highlight_colour || AppConfig.SITE_HEADER_DEFAULT_HIGHLIGHT_COLOUR,
                },
            },
        });

        return (

            <div>
                <Navbar className="global-header" style={styles.customHeaderBg} expand="lg">
                    <div className="logo-parent-div">
                        {siteHeader.logo_url
                        && (
                            <div style={styles.customLogoParentBg} className="customer-logo-parent-div">
                                <NavLink className="customer-logo-nav-link" to="/" tag={Link} onClick={this.toggleNav}>
                                    <img alt="Logo" className="customer-logo custom-logo" src={this.state.siteHeader.logo_url} />
                                </NavLink>
                            </div>
                        )}
                        <NavLink className="kuja-kuja-logo-nav-link" to="/" tag={Link} onClick={this.toggleNav}>
                            <img alt="Logo" className="customer-logo" src={this.state.siteHeader.kujakuja_logo_url} />
                        </NavLink>
                    </div>
                    <NavbarToggler onClick={this.toggleNav} />
                    <Collapse isOpen={isNavOpen} navbar>
                        <Nav className="ml-auto" navbar>
                            {AppConfig.DATA_STUDIO_GRAPH_URL
                            && (
                                <NavItem>
                                    <NavLink
                                        to="/graph"
                                        tag={Link}
                                        activeClassName="navbar_active"
                                        onClick={this.toggleNav}
                                    >
                                        {i18n.t("header.graph")}
                                    </NavLink>
                                </NavItem>
                            )}
                            {process.env.DISPLAY_IDEA ? ("") : (
                                <NavItem>
                                    <NavLink
                                        to="/ideafeed"
                                        tag={Link}
                                        activeClassName="navbar_active"
                                        onClick={this.toggleNav}
                                    >
                                        {i18n.t("header.ideaFeed")}
                                    </NavLink>
                                </NavItem>
                            )}
                            <NavItem>
                                <NavLink
                                    to="/actionfeed"
                                    tag={Link}
                                    activeClassName="navbar_active"
                                    onClick={this.toggleNav}
                                >
                                    {i18n.t("header.actionFeed")}
                                </NavLink>
                            </NavItem>
                            <NavItem>
                                <NavLink
                                    to="/datamap"
                                    tag={Link}
                                    activeClassName="navbar_active"
                                    onClick={this.toggleNav}
                                >
                                    {i18n.t("header.dataMap")}
                                </NavLink>
                            </NavItem>
                            <NavItem>
                                <NavLink
                                    to="/snapshots"
                                    tag={Link}
                                    activeClassName="navbar_active"
                                    onClick={this.toggleNav}
                                >
                                    {i18n.t("header.snapshots")}
                                </NavLink>
                            </NavItem>

                            <li className="nav-item">
                                <a className="nav-link" href={this.generateAdminUrl()}>
                                    {i18n.t("header.admin")}
                                </a>
                            </li>

                            {AppConfig.LOGIN_REQUIRED && isAuthenticated && (
                                <li className="nav-item">
                                    <a onClick={logout} className="nav-link" href="/login">Logout</a>
                                </li>
                            )}

                            <NavItem>
                                <Dropdown isOpen={isLngSelectDropdownOpen} toggle={this.toggleLngSelectDropdown}>
                                    <DropdownToggle
                                        caret
                                        tag="span"
                                        onClick={this.toggleLngSelectDropdown}
                                        data-toggle="dropdown"
                                        aria-expanded={isLngSelectDropdownOpen}
                                    >
                                        {selectedLanguageName}
                                    </DropdownToggle>
                                    <DropdownMenu>
                                        <div className="arrow-up" />
                                        {lngOptions
                                        .filter((lng) => { return !AppConfig.GCP_PROJECT_ID.includes("raiseyourvoice") || lng.value != 'ru' })
                                        .map((lng) => {
                                            return (
                                                <DropdownItem
                                                    key={lng.value}
                                                    onClick={changeLanguage}
                                                    value={lng.value}
                                                    active={selectedLanguage === lng.value}
                                                >
                                                    {lng.name}
                                                </DropdownItem>
                                            );
                                        })}
                                    </DropdownMenu>
                                </Dropdown>
                            </NavItem>
                        </Nav>
                    </Collapse>
                </Navbar>
            </div>

        );
    }
}
GlobalHeader.contextType = AuthContext;
