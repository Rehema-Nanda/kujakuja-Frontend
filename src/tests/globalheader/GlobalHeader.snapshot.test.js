import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import GlobalHeader from '../../components/GlobalHeader';

it('renders correctly', () => {
    const tree = renderer.create(
        <Router>
            <GlobalHeader/>
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
