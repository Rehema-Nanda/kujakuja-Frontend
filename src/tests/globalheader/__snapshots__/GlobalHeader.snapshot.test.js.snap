// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="global-header navbar navbar-expand-lg"
    style={
      Object {
        "borderTopColor": "#FFC300",
      }
    }
  >
    <div
      className="logo-parent-div"
    >
      <a
        aria-current="page"
        className="kuja-kuja-logo-nav-link nav-link active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="Logo"
          className="customer-logo"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "global-logo-dark.svg",
            }
          }
        />
      </a>
    </div>
    <button
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      style={Object {}}
    >
      <ul
        className="ml-auto navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/graph"
            onClick={[Function]}
          />
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/ideafeed"
            onClick={[Function]}
          />
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/actionfeed"
            onClick={[Function]}
          />
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/datamap"
            onClick={[Function]}
          />
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/snapshots"
            onClick={[Function]}
          />
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="http://admin.localhost/"
          />
        </li>
        <li
          className="nav-item"
        >
          <div
            className="dropdown"
            onKeyDown={[Function]}
          >
            <span
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle"
              data-toggle="dropdown"
              onClick={[Function]}
            >
              <span
                className="sr-only"
              >
                Toggle Dropdown
              </span>
            </span>
            <div
              aria-hidden={true}
              className="dropdown-menu"
              role="menu"
              tabIndex="-1"
            >
              <div
                className="arrow-up"
              />
              <button
                className="dropdown-item"
                onClick={[Function]}
                tabIndex="0"
                type="button"
                value="en"
              >
                English
              </button>
              <button
                className="dropdown-item"
                onClick={[Function]}
                tabIndex="0"
                type="button"
                value="es"
              >
                Espa√±ol
              </button>
              <button
                className="dropdown-item"
                onClick={[Function]}
                tabIndex="0"
                type="button"
                value="sw"
              >
                Swahili
              </button>
              <button
                className="dropdown-item"
                onClick={[Function]}
                tabIndex="0"
                type="button"
                value="ru"
              >
                Russian
              </button>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
